compiler/expressions.go

```diff
@@ -1112,6 +1112,12 @@ func (c *funcContext) translateImplicitConversion(expr ast.Expr, desiredType typ
 	}
 
 	exprType := c.p.TypeOf(expr)
+
+	// whenever expr has an underlying *types.Interface type we need to $copyInterfaceVal
+	if _, isIface := exprType.Underlying().(*types.Interface); isIface {
```

Comment:
> As far as I can see, the only place it is needed is for the receiver when calling a method on an interface (your test case)

This was my initial understanding as well. But I put the code here because I wasn't sure whether I'd considered all cases. And furthermore I reasoned to myself that whilst there may be unnecessary copying going on, everything would still be "correct" (I actually now think that conclusion on correctness is, ironically, incorrect, but it's now something of a moot point).

Having just given it some more thought I think we can, as you suggest, handle this via the receiver, save ourselves lots of unnecessary copying.

I'll do so in my next commit.